--- a/django_openstack/api.py
+++ b/django_openstack/api.py
@@ -332,7 +332,9 @@
     o = urlparse(url_for(request, 'glance'))
     LOG.debug('glance_api connection created for host "%s:%d"' %
                      (o.hostname, o.port))
-    return glance.client.Client(o.hostname, o.port)
+    return glance.client.Client(o.hostname,
+                                o.port,
+                                auth_tok=request.user.token)
 
 
 def admin_api(request):
